phoenix-core/src/main/java/org/apache/phoenix/execute/ClientScanPlan.java

```diff
@@ -106,12 +106,18 @@ public ExplainPlan getExplainPlan() throws SQLException {
     }
 
     @Override
-    public QueryPlan limit(Integer limit) {
-        if (limit == this.limit || (limit != null && limit.equals(this.limit)))
+    public QueryPlan limit(Integer limit, Integer offset) {
+        if (limit == this.limit || (limit != null && limit.equals(this.limit))) {
```

Comment:
I can see there's a big confusion here. Basically, offset and limit should be treated the same way. So here, it should be "if both the old limit and the old offset are same as as the new ones specified, we return this; otherwise we create a new object."